my-code/src/components/Search/index.js

```diff
@@ -0,0 +1,117 @@
+import React, { useState, useEffect, useRef } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+import { MdClose } from 'react-icons/md';
+import { Container, Input } from './styles';
+import { fetchData } from '../../services/api';
+import LoadingIndicator from '../LoadingIndicator';
+
+export default function Search({ isDisabled }) {
+  const ENTER_KEY = 'Enter';
+  const dispatch = useDispatch();
+  const perPage = 10;
+  const inputRef = useRef(null);
+
+  // states
+  const [isLoading, setIsLoading] = useState(false);
+  const [notFound, setNotFound] = useState(false);
+  const [searchStr, setSearchStr] = useState('');
+
+  // Redux state: get the movies list.
+  const movies = useSelector(state => {
+    return state.movieReducer;
+  });
+
+  useEffect(() => {
+    setSearchStr(movies.searchStr);
+  }, []);
+
+  /**
+   * Fetches data from the API.
+   * @param {integer} currentPage
+   */
+  async function getData(currentPage) {
```

Comment:
I would just update the function name to something more meaningful: `getMoveList` or `getSearchResult`. Reading this function I know what it does, but I don't know what kind of date it is.

Reply:
Also, make sure that currentPage has a default value.